src/ports/catalog/queries.ts

```diff
@@ -339,27 +339,27 @@ export const getMaxPriceWhere = (filters: CatalogFilters) => {
   } else if (filters.onlyListing) {
     return SQL`max_price <= ${filters.maxPrice}`
   }
-  return SQL`(max_price <= ${filters.maxPrice} OR (price <= ${filters.maxPrice} AND available > 0 AND search_is_store_minter = true))`
+  return SQL`(max_price <= ${filters.maxPrice} OR (price <= ${filters.maxPrice} AND available > 0 AND (search_is_store_minter = true OR search_is_marketplace_v3_minter = true)))`
 }
 
 export const getContractAddressWhere = (filters: CatalogFilters) => {
   return SQL`items.collection_id = ANY(${filters.contractAddresses})`
 }
 
 export const getOnlyListingsWhere = () => {
-  return SQL`(items.search_is_store_minter = false OR (items.search_is_store_minter = true AND available = 0)) AND listings_count > 0`
+  return SQL`((items.search_is_store_minter = false AND items.search_is_marketplace_v3_minter = false) OR (items.search_is_store_minter = true AND available = 0)) AND listings_count > 0`
 }
 
 export const getOnlyListingsWhereWithTrades = () => {
-  return SQL`(items.search_is_store_minter = false OR (items.search_is_store_minter = true AND available = 0)) AND (COALESCE(nfts_with_orders.orders_listings_count, 0) + COALESCE(offchain_orders.nfts_listings_count, 0)) > 0`
+  return SQL`((items.search_is_store_minter = false AND items.search_is_marketplace_v3_minter = false) OR (items.search_is_store_minter = true AND available = 0)) AND (COALESCE(nfts_with_orders.orders_listings_count, 0) + COALESCE(offchain_orders.nfts_listings_count, 0)) > 0`
```

Comment:
If we want only listing to be taken into consideration here, if I'm not wrong, we should consider checking:
1. When the item has not set as minter any of the two contracts.
2. When the item has set as minter the old contract and available is 0
3. When the item has set as minter the new contract and there aren't any item trades or available is 0
AND
1. There are open listings in the old contract
2. There are NFT trades.

```suggestion
return SQL`((available = 0 OR 
  (items.search_is_store_minter = false) OR
     (items.search_is_marketplace_v3_minter = false OR 
       (items.search_is_marketplace_v3_minter = true AND offchain_orders.count IS NULL)
      ))
  AND 
   (COALESCE(nfts_with_orders.orders_listings_count, 0) + COALESCE(offchain_orders.nfts_listings_count, 0)) > 0
```
I've used `offchain_orders.count` here, but it does not represent the number of item trades available. We could add another count to the `offchain_orders` CTE to count them.

Reply:
sounds good, using:
```
return SQL`((items.search_is_store_minter = false AND items.search_is_marketplace_v3_minter = false) OR (items.search_is_store_minter = true AND available = 0) OR (items.search_is_marketplace_v3_minter = true AND (COALESCE(offchain_orders.items_listings_count, 0) = 0 OR available = 0))) AND (COALESCE(nfts_with_orders.orders_listings_count, 0) + COALESCE(offchain_orders.nfts_listings_count, 0)) > 0`
```
and adding `COUNT(id) FILTER (WHERE status = 'open' and type = 'public_item_order') AS items_listings_count,` in the getTradesJoin